mantidproject/mantid
# Use package suffixes properly for macOS builds

**Description of work.**

This brings the macOS packaging inline with Windows/Linux where the nightly package is a separately
installable entitity to the main release packages. Both the package itself and program name are suffixed
along with using different icons in the launcher menus.

**To test:**

* Run `cmake -DENABLE_CPACK=ON -DCPACK_PACKAGE_SUFFIX=nightly` on your build to enable packaging
* Build with `ninja`
* Build a package `ninja package`. Note that on catalina you may have to remove PyCifRW from [requirements.txt](https://github.com/mantidproject/mantid/blob/55fe471b6514443fcd35a8e1b3e7b59e7696455d/installers/MacInstaller/requirements.txt) due to a bug in packaging. This issue has been fixed upstream but no release as come through yet.
* Install mantidnightly and see it can be used separate to a normal release without interference.

Fixes #27453 



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
